docs/Actions/apply-tokens.md

```diff
@@ -0,0 +1,165 @@
+
+Let's say we have the above template stored in \[EmailTemplate\]
+
+When will bring the template value will have to set it's value in a token that will contain tokens itself so we will need to set the recursion to 2 to replace the token to the template and the template to a plain token. See below example for Recursion set to 2:
+
+\[EmailTemplate\] -> Hi, \[User:FirstName\] \[User:LastName\]. Please contact me! -> Hi, John Doe. Please contact me!
+
+
+
+## `Examples`
+
+### `1. Calculations with Variable DoMath (Recursion =1)`
+
+By default the expression will only concatenate the variables and/or text from the token expression so in case we need to do some math Variables provides a token called DoMath that can solve this issue.
+
+
+```json
+
+{
+    "Title": "Apply Tokens",
+    "ActionType": "ApplyTokens",
+    "Parameters": {
+        "Tokens": [
+            {
+                "value": "[DoMath:Eval(expression=[User:UserId]*2)]",
+                "name": "DoubleOfUserId"
+            },
+            {
+                "value": "[DoMath:Eval(expression=(5*2)+2)]",
+                "name": "DoubleOfFivePlusTwo"
+            }
+        ],
+        "RecursionCount": "1"
+    }
+}
+​
+```
+
+### `2. Apply tokens inside a template (Recursion >1)`
+
+This example will show you how to replace tokens inside another token.
+
+```json
+
+{
+    "Title": "Execute Actions",
+    "ActionType": "ExecuteActions",
+    "Description": "Apply tokens inside a template",
+    "Parameters": {
+        "ActionList": [
+            {
+                "Title": "Apply Tokens",
+                "ActionType": "ApplyTokens",
+                "Parameters": {
+                    "Tokens": [
+                        {
+                            "value": "Hi, [User:FirstName] [User:LastName]. Please contact me!",
+                            "name": "EmailTemplate"
+                        }
+                    ],
+                    "RecursionCount": "1"
+                },
+                "$_uid": "action15910976849431312",
+                "$_isOpen": true,
+                "$_isLoaded": true,
+                "$_isFocus": true,
+                "Description": "Assign Template to Token"
+            },
+            {
+                "Title": "Apply Tokens",
+                "ActionType": "ApplyTokens",
+                "Parameters": {
+                    "Tokens": [
+                        {
+                            "value": "[EmailTemplate]",
+                            "name": "FinalTemplate"
+                        }
+                    ],
+                    "RecursionCount": "2"
+                },
+                "$_uid": "action1591097684943656",
+                "$_isOpen": true,
+                "$_isLoaded": true,
+                "$_isFocus": true,
+                "Id": -1,
+                "Description": "Apply tokens twice"
+            }
+        ]
+    }
+}
+
+```
+
+## `Frequently asked questions`

```

Comment:
I like the questions that were addressed. However, seems like many of them apply to any action, not just this one. So we can pull them on separate pages.
Also, for the rest, I would try to integrate them into the section they belong to. I think the FAQs are very useful, but not on a documentation page where all should have already been covered. I see FAQ more of a summary of an extensive documentation. We can discuss having a separate FAQ section in the documentation that will live at learn.plantanapp.com/faq.

Reply:
The reality is that all the questions are generic. I will remove them and move them to the FAQ page and make those questions generic to apply to all the Actions

Reply:
I made a section for the "questions" that are action-specific related